libraries/integration/Microsoft.Bot.Builder.Integration.NetCore/ApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Bot.Builder.Adapters;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using System;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Integration.NetCore
+{
+    public static class ApplicationBuilderExtensions
+    {
+        private static readonly JsonSerializer ActivitySerializer = JsonSerializer.Create();
+
+        public static IApplicationBuilder UseBotFramework(this IApplicationBuilder applicationBuilder)
+        {
+            var options = applicationBuilder.ApplicationServices.GetRequiredService<IOptions<BotFrameworkOptions>>().Value;
+
+            var botFrameworkAdapter = new BotFrameworkAdapter(options.ApplicationId, options.ApplicationPassword);
+
+            foreach (var middleware in options.Middleware)
```

Comment:
Same thing goes here as well ICredentialProvider instead of appId and password.

Reply:
Yup, #237.